services/worker-manager/src/provisioner.js

```diff
@@ -35,20 +44,61 @@ class Provisioner {
   async initiate() {
     await Promise.all(Object.values(this.providers).map(x => x.initiate()));
     await this.iterate.start();
+
+    this.pq = await consume({
+      client: this.pulseClient,
+      bindings: this.bindings,
+      queueName: 'workerTypeUpdates',
+    },
+    this.monitor.timedHandler('notification', this.onMessage.bind(this)),
+    );
   }
 
   /**
    * Terminate the Provisioner
    */
   async terminate() {
+    if (this.pq) {
+      await this.pq.stop();
+      this.pq = null;
+    }
     await this.iterate.stop();
     await Promise.all(Object.values(this.providers).map(x => x.terminate()));
   }
 
+  async onMessage({exchange, payload}) {
+    const {name, provider: providerName, previousProvider} = payload;
+    const workerType = await this.WorkerType.load({name});
+    const provider = this.providers[providerName]; // Always have a provider
+    switch (exchange.split('/').pop()) {
```

Comment:
I wonder if it would be better to have just separate handler functions for separate events?

Reply:
Ok, sounds good to me!